tests/accelerators/test_tpu_backend.py

```diff
@@ -257,3 +227,32 @@ def test_ddp_cpu_not_supported_on_tpus():
 
     with pytest.raises(MisconfigurationException, match="`accelerator='ddp_cpu'` is not supported on TPU machines"):
         Trainer(accelerator="ddp_cpu")
+
+
+@RunIf(tpu=True)
+def test_auto_weight_tying_tpus(tmpdir):
+
+    model = WeightSharingModule()
```

Comment:
Let's add nested modules to make sure it works properly.

Reply:
Updated!